domain/src/main/java/org/sagebionetworks/research/domain/interfaces/ObjectHelper.java

```diff
@@ -36,21 +36,10 @@
 import com.google.common.base.MoreObjects.ToStringHelper;
 
 public abstract class ObjectHelper {
-    // It is assumed that all classes that extend object helper are immutable. This is done in order
-    // to make storing a hash code possible.
-    protected final int hashCode;
-
-    /**
-     * Initializes the hashCode for this object. Subclasses should make sure to call this from all of their
-     * constructors.
-     */
-    public ObjectHelper() {
-        this.hashCode = this.hashCodeHelper().hash();
-    }
 
     @Override
     public int hashCode() {
-        return this.hashCode;
+        return hashCodeHelper().hash();
```

Comment:
I am concerned this will be slow. I don't think we want hash code to be creating an object every time it is called. What if instead we implemented some sort of singleton like pattern where we have Integer hashCode, and then we do
if (this.hashCode == null) {
this.hashCode = this.hashCodeHelper().hash();
}
return this.hashCode;

Reply:
That could work. Or how about having ObjectHelper instantiate the instance of HashCodeHelper as a final field?